ui/app/components/transit-key-actions.js

```diff
@@ -35,213 +49,141 @@ const TRANSIT_PARAMS = {
     if (resp && resp.data) {
       if (action === 'export' && resp.data.keys) {
         const { keys, type, name } = resp.data;
         resp.data.keys = { keys, type, name };
       }
-      props = assign({}, props, resp.data);
+      this.props = { ...this.props, ...resp.data };
     }
     if (options.wrapTTL) {
-      props = assign({}, props, { wrappedToken: resp.wrap_info.token });
+      this.props = { ...this.props, ...{ wrappedToken: resp.wrap_info.token } };
     }
-    if (!this.isDestroyed && !this.isDestroying) {
-      this.toggleProperty('isModalActive');
-      this.setProperties(props);
+    // open the modal
+    this.isModalActive = !this.isModalActive;
+    // verify doesn't trigger a success message
+    if (this.selectedAction !== 'verify') {
+      this.flashMessages.success(SUCCESS_MESSAGE_FOR_ACTION[action]);
     }
-    if (action === 'rotate') {
-      this.onRefresh();
-    }
-    this.triggerSuccessMessage(action);
-  },
+  }
+
+  @action updateProps() {
+    this.errors = null;
+    // There are specific props we want to carry over from the previous tab.
+    // Ex: carrying over this.props.context from the encrypt tab to the decrypt tab, but not carrying over this.props.plaintext.
+    // To do this, we make a new object that contains the old this.props key/values from the previous tab that we want to keep. We then merge that new object into the STARTING_TRANSIT_PROPS object to come up with our new this.props tracked property.
+    const transferredProps = PROPS_TO_KEEP[this.args.selectedAction].reduce(
+      (obj, key) => ({ ...obj, [key]: this.props[key] }),
+      {}
+    );
+    this.props = { ...STARTING_TRANSIT_PROPS, ...transferredProps };
+  }
 
   compactData(data) {
-    const type = this.key.type;
-    const isRSA = type === 'rsa-2048' || type === 'rsa-3072' || type === 'rsa-4096';
     return Object.keys(data).reduce((result, key) => {
-      if (key === 'signature_algorithm' && !isRSA) {
+      if (key === 'signature_algorithm' && !this.keyIsRSA) {
         return result;
       }
       if (data[key]) {
         result[key] = data[key];
       }
       return result;
     }, {});
-  },
-
-  actions: {
-    onActionChange(action) {
-      set(this, 'selectedAction', action);
-      this.checkAction();
-    },
-
-    onClear() {
-      this.resetParams(null, this.selectedAction);
-    },
-
-    clearParams(params) {
-      const arr = Array.isArray(params) ? params : [params];
-      arr.forEach((param) => this.set(param, null));
-    },
-
-    doSubmit(data, options = {}, maybeEvent) {
-      const event = options.type === 'submit' ? options : maybeEvent;
-      if (event) {
-        event.preventDefault();
+  }
+
+  @action toggleEncodeBase64() {
+    this.props.encodedBase64 = !this.props.encodedBase64;
+  }
+
+  @action clearSpecificProps(arrayToClear) {
+    arrayToClear.forEach((prop) => (this.props[prop] = null));
+  }
+
+  @task
+  @waitFor
+  *doSubmit(data, options = {}, maybeEvent) {
```

Comment:
Typically we clear/reset errors at the beginning of a submit task, should we add that here?